database/db.go

```diff
@@ -88,24 +86,41 @@ func New(ctx context.Context, creds database.DBCredentials, logger *logrus.Logge
 		return nil, err
 	}
 	return &DB{
-		DB:       db,
-		Skylinks: db.Collection(dbSkylinks),
-		Logger:   logger,
+		db,
+		logger,
 	}, nil
 }
 
+// Collection gets a handle for a collection with the given name configured with
+// the given CollectionOptions.
+func (db *DB) Collection(name string, opts ...*options.CollectionOptions) *mongo.Collection {
+	return db.staticDB.Collection(name, opts...)
+}
+
+// FindOneSkylink executes a find command on Skylinks collection and returns a
+// SingleResult for one document in the collection.
+func (db *DB) FindOneSkylink(ctx context.Context, filter interface{}, opts ...*options.FindOneOptions) *mongo.SingleResult {
+	return db.staticDB.Collection(collSkylinks).FindOne(ctx, filter, opts...)
+}
+
+// UpdateOneSkylink executes an update command on the Skylinks collection to
+// update at most one document in the collection.
+func (db *DB) UpdateOneSkylink(ctx context.Context, filter interface{}, update interface{}, opts ...*options.UpdateOptions) (*mongo.UpdateResult, error) {
```

Comment:
f/u

same thing here. Instead of a simple passthrough function, this should be more specific.

e.g. if you want to update the blocking state of a link this would be `UpdateSkylinkState(ctx, sl, state)`. Or if you use this method for 3 different things, add 3 different methods. Later each method will have its own unit test.

Reply:
F/U https://github.com/SkynetLabs/malware-scanner/issues/10